pages/glossary/semantic-role.md

```diff
@@ -9,4 +9,19 @@ A _semantic [role](https://www.w3.org/TR/wai-aria-1.1/#dfn-role)_ is a semantic
 
 Valid semantic roles are defined by standards. For web content and applications these are defined within the various [WAI-ARIA specifications](https://act-rules.github.io/glossary/#wai-aria-specifications).
 
-The semantic role of an element is its [explicit semantic role](#explicit-role) if it has any, otherwise, the [implicit semantic role](#implicit-role) is used.
+The _semantic role_ of an element is determined by the first of these cases that applies:
+
+1. **(Conflict)** If the element has an [explicit role][] which is either `none` or `presentation`, but the element is [included in the accessibility tree][] (or would be when it is not [hidden](https://www.w3.org/TR/wai-aria-1.2/#dfn-hidden)), then its _semantic role_ is its **[implicit role][]**.
+2. **(Explicit)** If the element has an [explicit role][], then its _semantic role_ is its [explicit role][].
```

Comment:
Here's a problem... Explicit / Implicit semantic role relies on the definition of semantic role. These are recursive definitions. We're going to need to solve that.

Reply:
Hmmm :thinking: 
Actually, the circular reference already exists in the current definitions…

---
My analysis of the circularity:
* there is some concept called "semantic role".
* there is some technical stuff called explicit/implicit role that are special case of the concept of semantic role.
* there is some technical stiff called "semantic role" which is define based on the explicit/implicit role and is thus also a case of the concept of semantic role.

Thus, the definition is not logically circular. This is more an unfortunate consequence of the fact that the concept and the technical thingie are named the same way…

This should be solved nonetheless.

---
In any case, given that this circularity already exists in the current definitions, and that I don’t think this PR is making things worse, I suggest to keep resolving that for another PR (keeping this one focused).